### DISCUSSION TITLE
Trust a Self-Signed Certificate in the Envoy API Gateway

### DISCUSSION THREAD

**Discussion by swageblock** [mia-platform/community#426] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

1.96.3

### Description

In our K8s configuration we are blocking egress traffic that is not proxied through the Envoy proxy.  In doing this we are aiming to shape our network more precisely for security and security reporting reasons.  We are needing encrypted communication to some legacy systems that utilize a custom ca.  We are needing to add this custom certificate to the list of trusted certs in the proxy.  What are the recommended ways of adding this certificate to the Envoy configuration? 

### Actual Outcome

Errors upon making http requests to the legacy systems utilizing a custom Ca.

### Expected Outcome

Trusted TLS connections between Mia workloads and workloads secured by a custom CA.

**Comment by davidebianchi** [comment, marked as answer]
Hi @swageblock! There are 2 different ways: one way is to modify the ca used by the service itself, the other is to configure the cluster configuration as showed [in this doc](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl).

It is possible to configure the clusters configuration through the advanced section, as showed [in this doc](https://docs.mia-platform.eu/docs/development_suite/api-console/advanced-section/api-gateway-envoy/extensions#clusters), otherwise it is possible to change the ca mounted by the service using the configmaps.
